pkg/deploy/strategy/recreate/recreate.go

```diff
@@ -33,6 +34,9 @@ type RecreateDeploymentStrategy struct {
 	until string
 	// getReplicationController knows how to get a replication controller.
 	getReplicationController func(namespace, name string) (*kapi.ReplicationController, error)
+	// listReplicationControllerEvents knows how to list all events related the
+	// given replication controller.
+	listReplicationControllerEvents func(rc *kapi.ReplicationController) (*kapi.EventList, error)
```

Comment:
The deployer code lags behind but we should definitely refactor it. @mfojtik mind having it as a follow-up?